common/tools/eslint-plugin-azure-sdk/src/rules/ts-use-cjs-polyfill.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @file Rule to require proper usage of ESM / CommonJS polyfills in TypeScript files.
+ *
+ */
+
+import { createRule } from "../utils/index.js";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+export default createRule({
+  name: "ts-use-cjs-polyfill",
+  meta: {
+    type: "problem",
+    docs: {
+      description: "Require a proper polyfill when using CommonJS concepts in ESM files.",
+    },
+    messages: {
+      usePolyfill:
+        "__dirname and __filename should not be used in ESM files. Use the polyfill documented in https://github.com/Azure/azure-sdk-for-js/wiki/Dual-emitting-using-tshy",
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      Identifier: (node): void => {
+        if (node.name === "__dirname" || node.name === "__filename") {
+          // tshy will handle commonJS / ESM polyfills when building based on file extension.
+          // If the file ends in -cjs.cts, it is a CommonJS file and could use commonJS concepts.
+          if (!context.filename.endsWith("-cjs.cts")) {
```

Comment:
is it -cjs or -commonjs? I see the dist folder named commonjs: https://github.com/Azure/azure-sdk-for-js/blob/54134eec16797cc8192efea2fa6f0c2b98725c25/sdk/identity/identity/package.json#L143. Also, since this rule applies everywhere not just under src, perhaps we can simplify it to be looking for just the .cts extension.

Reply:
It is -cjs.cts (which is important for tshy to pick up the correct polyfill https://github.com/[isaacs/tshy](https://github.com/isaacs/tshy?tab=readme-ov-file#commonjs-dialect-polyfills)?tab=readme-ov-file#commonjs-dialect-polyfills)

I am actually working on limiting it to just `src` now - there's no need to validate this in tests so stay tuned for another commit 👍